force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,11 +138,39 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        
+        List<Account> accounts =[SELECT Id, Name                                    
+                                FROM Account 
+                                WHERE Name = :accountName];
+        Account acct = new Account();                        
+        if (!accounts.isempty()){                   //checks for existing account
+            acct = accounts[0];            
+        } else {                                    // otherwise, create a new Account
+            acct.Name = accountName;
+        }   
+        upsert acct;                        		// Upsert the Account
+                                
+        List<Opportunity> oppList = new List<Opportunity>([SELECT Id, Name FROM Opportunity WHERE Name IN :oppNames]);
+        Set<String>oppNamesSet = new Set<String> (oppNames);
+        if (!oppList.isempty()){
```

Comment:
You dont really need to check here if this isEmpty because trying to loop an empty list will just loop 0 times.